### PR TITLE
media-video/haruna: new ebuild

### PR DESCRIPTION
**Pull request by oniGino**
Add haruna video player

### CODE REVIEW

**Review by a17r — CHANGES_REQUESTED** [review]
Should use kde.org.eclass and ecm.eclass instead.
Tags are no releases. There will be a proper tarball on kde.org mirrors to download from.

**Comment by oniGino** [comment]
> Should use kde.org.eclass and ecm.eclass instead. Tags are no releases. There will be a proper tarball on kde.org mirrors to download from.

1. fixed eclass request
2. they post flatpaks and appimages but no tarballs (unfortunately )

File: media-video/haruna/haruna-0.12.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+RDEPEND="
+	dev-qt/qtconcurrent:5
+	dev-qt/qtdeclarative:5
+	dev-qt/qtcore:5
+	dev-qt/qtdbus:5
+	dev-qt/qtgui:5
+	dev-qt/qtquickcontrols:5
+	dev-qt/qtquickcontrols2:5
+	dev-qt/qtxml:5
+	dev-qt/qtx11extras:5
```
Comment (a17r) [inline media-video/haruna/haruna-0.12.3.ebuild:25]: keep them sorted please. dev-qt/qtcore can be dropped, it is always provided by ecm.eclass.

File: media-video/haruna/haruna-0.12.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit xdg kde.org ecm
```
Comment (a17r) [inline media-video/haruna/haruna-0.12.3.ebuild:6]: xdg can be dropped, done by ecm.eclass.

File: media-video/haruna/haruna-0.12.3.ebuild [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	kde-frameworks/ki18n:5
+	kde-frameworks/kiconthemes:5
+	kde-frameworks/kio:5
+	kde-frameworks/kirigami:5
+	kde-frameworks/qqc2-desktop-style:5
+	media-video/mpv[libmpv]
+	youtube? ( net-misc/yt-dlp )"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	>=kde-frameworks/extra-cmake-modules-5.112.0:0
```
Comment (a17r) [inline media-video/haruna/haruna-0.12.3.ebuild:39]: ECM can be dropped, it is always provided by ecm.eclass.

**Comment by oniGino** [comment]
> > Should use kde.org.eclass and ecm.eclass instead. Tags are no releases. There will be a proper tarball on kde.org mirrors to download from.
> 
>     1. fixed eclass request
> 
>     2. they post flatpaks and appimages but no tarballs (unfortunately )

my mistake I found it, thank you.

**Review by a17r — CHANGES_REQUESTED** [review]
If 5.112.0 is the minimum for ECM, then all KDE Frameworks must have this minimum dep.

Our standardised method for enforcing this is adding `KFMIN=5.112.0` pre-inherit and change all `kde-frameworks/*` lines to use it. `ecm.eclass` will also apply `KFMIN`.

File: media-video/haruna/haruna-0.12.3.ebuild [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	sys-devel/gettext"
+
+src_configure() {
+	local mycmakeargs=(
+		-DKDE_INSTALL_DOCBUNDLEDIR="${EPREFIX}/usr/share/help"
+	)
+	cmake_src_configure
+}
```
Comment (a17r) [inline media-video/haruna/haruna-0.12.3.ebuild:47]: can be dropped since using `ecm.eclass` now.

**Review by a17r — CHANGES_REQUESTED** [review]

File: media-video/haruna/haruna-0.12.3.ebuild [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+	kde-frameworks/kconfig:5
+	kde-frameworks/kcoreaddons:5
+	kde-frameworks/kfilemetadata:5
+	kde-frameworks/ki18n:5
+	kde-frameworks/kiconthemes:5
+	kde-frameworks/kio:5
+	kde-frameworks/kirigami:5
+	kde-frameworks/qqc2-desktop-style:5
+	media-video/mpv[libmpv]
+	youtube? ( net-misc/yt-dlp )"
```
Comment (a17r) [inline media-video/haruna/haruna-0.12.3.ebuild:36]: This is not connected to any kind of build option? How does it work, then?

**Review by a17r — CHANGES_REQUESTED** [review]

File: media-video/haruna/haruna-0.12.3.ebuild [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+DEPEND="${RDEPEND}"
+BDEPEND="
+	sys-devel/gettext"
+
+src_configure() {
+	local mycmakeargs=(
+		$(cmake_use_find_package youtube Ytdlp)
+	)
+	ecm_src_configure
+	cmake_src_configure
```
Comment (a17r) [inline media-video/haruna/haruna-0.12.3.ebuild:48]: You call `ecm_src_configure` only, not `cmake_src_configure`.

File: media-video/haruna/haruna-0.12.3.ebuild [hunk 7]
```diff
@@ -0,0 +1,49 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit kde.org ecm
+
+KFMIN=5.112.0
+QTMIN=5.15.11
```
Comment (a17r) [inline media-video/haruna/haruna-0.12.3.ebuild:9]: `KFMIN` and `QTMIN` shall be set **pre**-inherit

File: media-video/haruna/haruna-0.12.3.ebuild [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+DESCRIPTION="Video player built with Qt/QML on top of libmpv"
+HOMEPAGE="https://invent.kde.org/multimedia/haruna"
+SRC_URI="https://download.kde.org/stable/${PN}/${P}.tar.xz"
+LICENSE="GPL-3"
```
Comment (a17r) [inline media-video/haruna/haruna-0.12.3.ebuild:14]:
```suggestion
SRC_URI="https://download.kde.org/stable/${PN}/${P}.tar.xz"

LICENSE="GPL-3"
```

File: media-video/haruna/haruna-0.12.3.ebuild [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+KEYWORDS="~amd64 ~x86"
```
Comment (a17r) [inline media-video/haruna/haruna-0.12.3.ebuild:15]: - did you test this on ~x86?

File: media-video/haruna/haruna-0.12.3.ebuild [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+SLOT="0"
+IUSE="youtube"
```
Comment (a17r) [inline media-video/haruna/haruna-0.12.3.ebuild:17]: correct sorting per https://github.com/gentoo/gentoo/blob/master/skel.ebuild:
- LICENSE
- SLOT
- KEYWORDS
- IUSE

**Review by a17r — CHANGES_REQUESTED** [review]

**Pull request closed without merging**